rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -1582,4 +1590,18 @@ public void addEmbeddedReleaseDetails(HalResource<ClearingRequest> halClearingRe
         halClearingRequest.addEmbeddedResource("openRelease", openReleaseCount);
         halClearingRequest.addEmbeddedResource("totalRelease", totalReleaseCount);
     }
+
+    public ClearingRequest updateCRSize(ClearingRequest clearingRequest, Project project, User sw360User) throws TException {
+        int openReleaseCount = SW360Utils.getOpenReleaseCount(project.getReleaseClearingStateSummary());
+        ClearingRequestSize currentSize = SW360Utils.determineCRSize(openReleaseCount);
+        ClearingRequestSize initialSize = clearingRequest.getClearingSize();
+        if(initialSize == null) return clearingRequest;
+        if(!initialSize.equals(ClearingRequestSize.VERY_LARGE)) {
+            int limit = CLEARING_REQUEST_SIZE_MAP.get(initialSize);
+            if(openReleaseCount > limit){
```

Comment:
@akshitjoshii Instead of putting limit map at 2 places (here and in `SW360Utils::determineCRSize()`), can we instead compare `currentSize` and `initialSize`? They will not be equal if there is an size update and thus the DB should be updated, as per my understanding.

This will help reduce number of places to be updated if the size limits are updated in future.